src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,230 @@
+import com.aws.greengrass.lifecyclemanager.Kernel;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.mqttclient.MqttClient;
+import com.aws.greengrass.security.SecurityService;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import com.aws.greengrass.util.Pair;
+import lombok.AccessLevel;
+import lombok.Setter;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.crt.mqtt.MqttException;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import software.amazon.awssdk.services.greengrassv2data.model.ListThingGroupsForCoreDeviceRequest;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ConnectivityValidator {
+    private static final Logger logger = LogManager.getLogger(ConnectivityValidator.class);
+    public static final String clientIdSuffix = "#validation";
+    private final MqttClient mqttClient;
+    private final GreengrassServiceClientFactory factory;
+    private final SecurityService securityService;
+    @Setter(AccessLevel.PACKAGE)  // for unit tests
+    private DeviceConfiguration deploymentConfiguration;
+
+    /**
+     * Constructor.
+     *
+     * @param kernel           kernel to get current config and context from
+     * @param context          temporary context used for deployment configuration
+     * @param mqttClient       Used to create AWS Mqtt Connection Client
+     * @param factory          Used by integration tests to set mock
+     * @param deploymentConfig Map of the configs to be merged in and validated against
+     * @param timestamp        time stamp of the deployment
+     */
+    public ConnectivityValidator(Kernel kernel, Context context, MqttClient mqttClient,
+                                 GreengrassServiceClientFactory factory, Map<String, Object> deploymentConfig,
+                                 long timestamp) {
+        this.mqttClient = mqttClient;
+        this.factory = factory;
+        securityService = kernel.getContext().get(SecurityService.class);
+
+        Configuration config = new Configuration(context);
+        // Copy the current device configuration Map to preserve time stamps
+        config.copyFrom(kernel.getConfig());
+        // Attempt to merge the deployment configs using the deployment time stamp
+        config.mergeMap(timestamp, deploymentConfig);
+        try {
+            config.waitConfigUpdateComplete();
+        } catch (InterruptedException e) {
+            logger.atError().log("Wait for deployment config update complete was interrupted");
```

Comment:
get more specific in this message. Why should we care about this, what does it mean, what should a user do if they see this message?